src/web/handlers.rs

```diff
@@ -418,15 +394,12 @@ pub fn get_bso(bso_req: BsoRequest) -> impl Future<Output = Result<HttpResponse,
             id: bso_req.bso,
         })
         .map_err(From::from)
-        .map(
-            |result: std::result::Result<std::option::Option<GetBso>, Error>| {
-                let result = result.expect("Could not get result in get_bso");
-                result.map_or_else(
-                    || Ok(HttpResponse::NotFound().finish()),
-                    |bso| Ok(HttpResponse::Ok().json(bso)),
-                )
-            },
-        )
+        .map_ok(|result: std::option::Option<GetBso>| {
```

Comment:
```suggestion
        .map_ok(|result: Option<GetBso>| {
```

and also all the `std::collections::HashMap` -> `HashMap` (there's already a use statement for it)

Reply:
On second thought: 

```suggestion
        .map_ok(|result| {
```

..I doubt we need any of these type annotations any longer (the original version didn't)